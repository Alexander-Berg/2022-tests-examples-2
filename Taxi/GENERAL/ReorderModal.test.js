/* eslint-disable max-len */
import React from 'react';
import {ConfirmModal} from '@yandex-taxi/corp-core/modal';

import {shallow} from 'enzyme';
import {identity} from 'lodash';

import {ReorderModal} from './ReorderModal';

describe('ReorderModal', () => {
    it.only('не позволяет нажать кнопку заказа дважды, но разблокирует её после заказа', async () => {
        let promise;
        const onOKClick = jest.fn(() => (promise = Promise.resolve()));
        const onCancelRequest = jest.fn();
        const wrapper = shallow(
            <ReorderModal onOKClick={onOKClick} onCloseRequest={onCancelRequest} t={identity} />,
        );
        expect(wrapper).toMatchSnapshot();
        wrapper.find(ConfirmModal).prop('onOKClick')();
        expect(onOKClick).toHaveBeenCalled();
        expect(wrapper.state().inProgress).toBeTruthy();
        expect(wrapper.find(ConfirmModal).prop('OKDisabled')).toBe(true);
        await promise;
        expect(wrapper.find(ConfirmModal).prop('onOKClick')).not.toBeUndefined();
    });
});
