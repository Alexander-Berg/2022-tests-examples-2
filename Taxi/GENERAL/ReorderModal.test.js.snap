// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReorderModal не позволяет нажать кнопку заказа дважды, но разблокирует её после заказа 1`] = `
<Component
  OKText="modal.order.price.changed.confirm"
  className="ReorderModal"
  onCancelClick={[MockFunction]}
  onCloseRequest={[MockFunction]}
  onOKClick={[Function]}
  title="modal.order.price.changed.title"
>
  <Section
    padding="l"
    theme="white"
  >
    <div
      className="amber-typography__body2"
    >
      <Connect(withProps(OrderEstimate))
        t={[Function]}
      />
    </div>
  </Section>
</Component>
`;
